lib/ably/realtime/channel.rb

```diff
@@ -179,15 +179,28 @@ def presence
         @presence
       end
 
-      # Return the message history of the channel
+      # Return the message history of the channel.
+      #
+      # Once attached to a channel, you can retrieve messages published on the channel before the
+      # channel was attached with the option <tt>:end => :before_attach</tt>.  This is very useful for
+      # developers who wish to display both new real-time messages as well as historical messages with
+      # the guarantee that no messages have been missed.
+      #
       #
       # @param (see Ably::Rest::Channel#history)
       # @option options (see Ably::Rest::Channel#history)
+      # @option options [Integer,Time,Symbol]   :end        Time or millisecond since epoch, or +:before_attach+ for all messages published before this channel has been attached.
       #
       # @yield [Ably::Models::PaginatedResource<Ably::Models::Message>] An Array of {Ably::Models::Message} objects that supports paging (#next_page, #first_page)
       #
       # @return [Ably::Util::SafeDeferrable]
       def history(options = {}, &callback)
+        if options[:end] == :before_attach
+          raise ArgumentError, 'option :before_attach cannot be specified if the channel is not attached' unless attached?
```

Comment:
There is no implicit `attach` because it's perfectly valid to ask for history on a channel you are not attached to and have no intention of attaching to.  Added to this, asking for "Give me everything before I attached" when you're not attached is simply wrong.  You shouldn't be asking for the history of a future event.